### PR TITLE
Daemon

### PR DESCRIPTION
**Pull request by MarnixCroes**

### CODE REVIEW

**Comment by MarnixCroes** [comment]
wait for merge until release?

File: docs/using-wasabi/Daemon.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  "title": "Headless Wasabi Daemon",
+  "description": "A guide on how to use the light weight headless Wasabi daemon. This is the Wasabi documentation, an archive of knowledge about the open-source, non-custodial and privacy-focused Bitcoin wallet for desktop."
+}
+---
+
+# Headless Wasabi Daemon
+
+The default of how to interact with your Wasabi wallet is the graphical user interface.
+There is also a headless daemon where you do not run a resource-intensive GUI, but only the command line interface.
+This daemon is especially useful for power users mixing bitcoin in the backend of their servers.
```
Comment (lontivero) [inline docs/using-wasabi/Daemon.md:12]: This is an opinion and to me is wrong. In fact, the final goal should be to have the daemon running on every desktop.

File: docs/using-wasabi/Daemon.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+# Headless Wasabi Daemon
+
+The default of how to interact with your Wasabi wallet is the graphical user interface.
+There is also a headless daemon where you do not run a resource-intensive GUI, but only the command line interface.
+
+The desktop app (GUI) is built on top of the daemon.
+Meaning that command line arguments/variables can also be used to configure the desktop app.
+
+---
```
Comment (yahiheb) [inline docs/using-wasabi/Daemon.md:16]:
```suggestion
```

**Review by yahiheb — CHANGES_REQUESTED** [review]

**Review by yahiheb — COMMENTED** [review]
Shouldn't we add some available commands and how to use the daemon?
Like we had before removing it here https://github.com/zkSNACKs/WasabiDoc/pull/1232/files

**Comment by MarnixCroes** [comment]
still a WIP

currently I have not mentioned `Environment variables` anywhere, should this be mentioned in the docs here? Does it add anything for the normal user? wdty @lontivero ?

**Comment by lontivero** [comment]
Environment variables have the same importance than command line switches. For example, these two are equivalent:

```bash
$ wassabee --network=main
```

```bash
$ WASABI_NETWORK=main wassabee
```

However, there are many cases where environment variables are much more convenient. Imagine you want to use a different DATADIR in your environment, how do you do it? Well, you just set the `WASABI_DATADIR` environment variable to wherever you want and that does  the trick. For example:

```
WASABI_DATADIR=Z:/encrypted-folder/wasabi
```

It is impossible to do this using config file and it is not so convenient to do it using command line switches.

**Comment by MarnixCroes** [comment]
> Is this PR finished @MarnixCroes ?

I need to add how to run on Win and mac as, afaik, you cannot just do `wassabeed` from terminal.
other than that yes.

File: docs/using-wasabi/Daemon.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+| "JsonRpcUser": "" | --jsonrpcuser="" | WASABI-JSONRPCUSER="" |
+| "JsonRpcPassword": "" | --jsonrpcpassword="" | WASABI-JSONRPCPASSWORD="" |
+| "JsonRpcServerPrefixes":["http://127.0.0.1:37128/", "http://localhost:37128/"] | --jsonrpcserverprefixes=["http://127.0.0.1:37128/", "http://localhost:37128/"] | WASABI-JSONRPCSERVERPREFIXES=["http://127.0.0.1:37128/", "http://localhost:37128/"] |
+| "DustThreshold": "0.00005" | --dustthreshold=0.00005 | WASABI-DUSTTHRESHOLD=0.00005 |
+| "EnableGpu": true | --enablegpu=true | WASABI-ENABLEGPU=true |
+| "CoordinatorIdentifier": "CoinJoinCoordinatorIdentifier" | --coordinatoridentifier="coinjoincoordinatoridentifier" | WASABI-COORDINATORIDENTIFIER="coinjoincoordinatoridentifier" |
+
+### Daemon only configurations
+
+There are a few special switches that are not present in the `Config.json` file and are only available using the command line:
```
[the remark this answers is not in the document]
Comment (MarnixCroes) [reply docs/using-wasabi/Daemon.md:57]: done

File: docs/using-wasabi/Daemon.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+| "JsonRpcServerPrefixes":["http://127.0.0.1:37128/", "http://localhost:37128/"] | --jsonrpcserverprefixes=["http://127.0.0.1:37128/", "http://localhost:37128/"] | WASABI-JSONRPCSERVERPREFIXES=["http://127.0.0.1:37128/", "http://localhost:37128/"] |
+| "DustThreshold": "0.00005" | --dustthreshold=0.00005 | WASABI-DUSTTHRESHOLD=0.00005 |
+| "EnableGpu": true | --enablegpu=true | WASABI-ENABLEGPU=true |
+| "CoordinatorIdentifier": "CoinJoinCoordinatorIdentifier" | --coordinatoridentifier="coinjoincoordinatoridentifier" | WASABI-COORDINATORIDENTIFIER="coinjoincoordinatoridentifier" |
+
+### Non-Config file configurations
+
+There are a few special switches that are not present in the `Config.json` file and are only available using the command line:
+
+ Switch | Command Line| Environment variable|
```
Comment (yahiheb) [inline docs/using-wasabi/Daemon.md:59]:
```suggestion
| Switch | Command Line| Environment variable|
```

File: docs/using-wasabi/Daemon.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+## Examples
+
+Run Wasabi Daemon and connect to the testnet Bitcoin network.
+
+```bash
+$ wassabeed --network=testnet
+```
+
+Run Wasabi and connect to the testnet Bitcoin network with Tor disabled and accept JSON RPC calls. Store everything in `$HOME/temp/wasabi-1`.
```
Comment (yahiheb) [inline docs/using-wasabi/Daemon.md:108]:
```suggestion
Run Wasabi and connect to the testnet Bitcoin network with Tor disabled and accept JSON RPC calls and ignore P2P transactions.
Store everything in `$HOME/temp/wasabi-1`.
```

**Review by yahiheb — CHANGES_REQUESTED** [review]

File: docs/using-wasabi/Daemon.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+| Ignore P2P transactions | --blockonly=true | WASABI-BLOCKONLY=true |
+| The level of detail used during logging | --loglevel=trace | WASABI-LOGLEVEL=trace |
+| The path to the directory used during runtime | --datadir="$HOME/temp/wasabi-1" | WASABI-DATADIR="$HOME/temp/wasabi-1" |
+| Open wallet "WalletName" | --wallet=WalletName | WASABI--WALLET=WalletName |
+
+## How to run
+
+### If the package is installed
+
+Depending on your operating system, open the command line and execute:
```
[the remark this answers is not in the document]
Comment (MarnixCroes) [reply docs/using-wasabi/Daemon.md:70]: done 074b818c0

**Review by MaxHillebrand — DISMISSED** [review]

File: docs/using-wasabi/Daemon.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+#### Windows
+
+```bash
+cd C:\Program Files\WasabiWallet
+wassabeed --wallet=WalletName --jsonrpcserverenabled=true
+```
+
+### If building from source code
+
+Open the terminal, navigate to the WalletWasabi.Daemon directory (inside the cloned repository) and execute the desired commands.
```
Comment (thibistaken) [inline docs/using-wasabi/Daemon.md:94]: Maybe add that you can use verbose log parameter `--loglevel=debug`?

Comment (MaxHillebrand) [reply docs/using-wasabi/Daemon.md:94]: See line 62

File: docs/using-wasabi/Daemon.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+[[toc]]
+
+## Introduction
+
+The default of how to interact with your Wasabi wallet is the graphical user interface (GUI).
+There is also a headless daemon where you do not run a resource-intensive GUI, but only the command line interface.
+Running the daemon minimizes the usage of resources (CPU, GPU, Memory, Bandwidth) with the goal of making it more suitable for running all the time in the background.
+The [RPC interface](/using-wasabi/RPC.md) can be used to interact with the Daemon.
+
+The daemon is included in the Wasabi package starting from version 2.0.4 and can be launched using the command line.
```
Comment (yahiheb) [inline docs/using-wasabi/Daemon.md:19]:
```suggestion
The daemon is included in the Wasabi package starting from version [2.0.4](https://github.com/zkSNACKs/WalletWasabi/releases/tag/v2.0.4) and can be launched using the command line.
```

File: docs/using-wasabi/Daemon.md (hunk 4 above)
Comment (yahiheb) [reply docs/using-wasabi/Daemon.md:59]: For consistency.

**Review by yahiheb — CHANGES_REQUESTED** [review]

File: docs/using-wasabi/Daemon.md (hunk 4 above)
Comment (yahiheb) [reply docs/using-wasabi/Daemon.md:59]: Update this for consistency.

**Pull request merged**